lib/open_food_network/proxy_order_syncer.rb

```diff
@@ -17,29 +17,36 @@ def initialize(subscriptions)
     end
 
     def sync!
-      return sync_all! if @subscriptions
+      return sync_subscriptions! if @subscriptions
+
       return initialise_proxy_orders! unless @subscription.id
 
-      create_proxy_orders!
-      remove_orphaned_proxy_orders!
+      sync_subscription!
     end
 
     private
 
-    def sync_all!
+    def sync_subscriptions!
       @subscriptions.each do |subscription|
         @subscription = subscription
```

Comment:
It took me a minute to find this line to understand that your change is correct here. It should really be `sync_subscription!(subscription)` instead of a leaking ivar. Out of scope here though.

Reply:
yeah, the use of these ivars is a bit weird in this class...